src/view/UploadView.java

```diff
@@ -98,15 +101,169 @@ public void actionPerformed(ActionEvent evt) {
+                new KeyListener() {
+                    @Override
+                    public void keyTyped(KeyEvent e) {
+                        UploadRecipeState currentState = uploadRecipeViewModel.getState();
+                        String text = nutritionInputField.getText() + e.getKeyChar();
+                        currentState.setNutrition(text);
+                        uploadRecipeViewModel.setState(currentState);
+                    }
+
+                    @Override
+                    public void keyPressed(KeyEvent e) {
+                    }
+
+                    @Override
+                    public void keyReleased(KeyEvent e) {
+                    }
+                });
+
+        instructionsInputField.addKeyListener(
+                new KeyListener() {
+                    @Override
+                    public void keyTyped(KeyEvent e) {
+                        UploadRecipeState currentState = uploadRecipeViewModel.getState();
+                        String text = instructionsInputField.getText() + e.getKeyChar();
+                        currentState.setInstructions(text);
+                        uploadRecipeViewModel.setState(currentState);
+                    }
+
+                    @Override
+                    public void keyPressed(KeyEvent e) {
+                    }
+
+                    @Override
+                    public void keyReleased(KeyEvent e) {
+                    }
+                });
+
+
+
+
+
+
+        imageInputField.addKeyListener(
+                new KeyListener() {
+                    @Override
+                    public void keyTyped(KeyEvent e) {
+                        UploadRecipeState currentState = uploadRecipeViewModel.getState();
+                        String text = imageInputField.getText() + e.getKeyChar();
+                        currentState.setImage(text);
+                        uploadRecipeViewModel.setState(currentState);
+                    }
+
+                    @Override
+                    public void keyPressed(KeyEvent e) {
+                    }
+
+                    @Override
+                    public void keyReleased(KeyEvent e) {
+                    }
+                });
+
+
+
+
+
+
+        recipeurlInputField.addKeyListener(
+                new KeyListener() {
+                    @Override
+                    public void keyTyped(KeyEvent e) {
+                        UploadRecipeState currentState = uploadRecipeViewModel.getState();
+                        String text = recipeurlInputField.getText() + e.getKeyChar();
+                        currentState.setRecipeurl(text);
+                        uploadRecipeViewModel.setState(currentState);
+                    }
+
+                    @Override
+                    public void keyPressed(KeyEvent e) {
+                    }
+
+                    @Override
+                    public void keyReleased(KeyEvent e) {
+                    }
+                });
+
+
+
+
+
```

Comment:
So in this class, there is only the methods? where is the constructor?

Reply:
I have wrote the constructor but it was hidden